ci/flaky_test/process_xml.py

```diff
@@ -0,0 +1,136 @@
+    log_file_path += token
+  log_file_path += ".log"
+
+  tree = ET.parse(file)
+
+  root = tree.getroot()
+  ret = ""
+
+  # This loop is dependent on the structure of xml file emitted for test runs.
+  # Should this change in the future, appropriate adjustments need to made.
+  for testsuite in root:
+    if (testsuite.attrib['failures'] != '0'):
+      for testcase in testsuite:
+        for failure_msg in testcase:
+          if (testcase.attrib['name'], testsuite.attrib['name']) not in visited:
+            ret += "-----------------------Flaky Testcase: {} in TestSuite: {} -----------------------\n".format(
+                testcase.attrib['name'], testsuite.attrib['name'])
+            ret += log_file_path + "\n" + failure_msg.text + "\n"
+            visited.add((testcase.attrib['name'], testsuite.attrib['name']))
+  return ret
+
+
+def processFindOutput(f, problematic_files):
+  for line in f:
+    lineList = line.split('/')
+    filepath = ""
+    for i in range(len(lineList)):
+      if i >= len(lineList) - 2:
+        break
+      filepath += lineList[i] + "/"
+    filepath += "test.xml"
+    problematic_files[filepath] = line.strip('\n')
+
+
+# Prints out helpful information on the run using Git.
+# Should Git changes the output of the used commands in the future,
+# this will likely need adjustments as well.
+def getGitInfo(CI_TARGET):
+  ret = ""
+  os.system("git remote -v > ${TMP_OUTPUT_PROCESS_XML}")
+  os.system("git describe --all >> ${TMP_OUTPUT_PROCESS_XML}")
+  os.system("git show >> ${TMP_OUTPUT_PROCESS_XML}")
+  f = open(os.environ['TMP_OUTPUT_PROCESS_XML'], 'r+')
+  # Fetching the URL from predefined env variable
+  envoy_link = os.environ["REPO_URI"]
+  for line in [next(f) for x in range(6)]:
+    if line.split('/')[0] == 'remotes':
+      for token in line.split('/')[1:-1]:
+        envoy_link += '/' + token
+    ret += line
+
+  ret += "link for additional content: " + envoy_link + " \n"
+  ret += "azure build URI: " + os.environ["BUILD_URI"] + " \n"
+  if CI_TARGET != "":
+    ret += "In " + CI_TARGET + " build\n"
+  return ret
+
+
+if __name__ == "__main__":
+  CI_TARGET = ""
+  if len(sys.argv) == 2:
+    CI_TARGET = sys.argv[1]
+  output_msg = "``` \n"
+  has_flaky_test = False
+
+  if os.getenv("TEST_TMPDIR") and os.getenv("REPO_URI") and os.getenv("BUILD_URI"):
+    os.environ["TMP_OUTPUT_PROCESS_XML"] = os.getenv("TEST_TMPDIR") + "/tmp_output_process_xml.txt"
+  else:
+    print("set the env variables first")
+    sys.exit(0)
+  output_msg += getGitInfo(CI_TARGET)
+
+  if CI_TARGET == "MacOS":
+    os.system('find ${TEST_TMPDIR}/ -name "attempt_*.xml" > ${TMP_OUTPUT_PROCESS_XML}')
+  else:
+    os.system(
+        'find ${TEST_TMPDIR}/**/**/**/**/bazel-testlogs/ -name "attempt_*.xml" > ${TMP_OUTPUT_PROCESS_XML}'
+    )
+
+  f = open(os.environ['TMP_OUTPUT_PROCESS_XML'], 'r+')
+  if f.closed:
+    print("cannot open {}".format(os.environ['TMP_OUTPUT_PROCESS_XML']))
+
+  problematic_files = {}
```

Comment:
nit: add comment about what problematic_files holds or doc string for processFindOutput